adapters/ix/ix.go

```diff
@@ -359,3 +508,104 @@ func BuildIxDiag(request *openrtb2.BidRequest) error {
 	}
 	return nil
 }
+
+// moves sid from imp[].ext.bidder.sid to imp[].ext.sid
+func moveSid(imp *openrtb2.Imp) error {
+	if imp.Ext == nil || len(imp.Ext) == 0 {
+		return nil
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return err
+	}
+
+	var ixExt openrtb_ext.ExtImpIx
+	if err := json.Unmarshal(bidderExt.Bidder, &ixExt); err != nil {
+		return err
+	}
+
+	if ixExt.Sid != "" {
+		var m map[string]interface{}
+		if err := json.Unmarshal(imp.Ext, &m); err != nil {
+			return err
+		}
+		m["sid"] = ixExt.Sid
+		ext, err := json.Marshal(m)
+		if err != nil {
+			return err
+		}
+		imp.Ext = ext
+	}
+	return nil
+}
+
+func setFeatureToggles(a *IxAdapter, ext *json.RawMessage) {
+	if *ext == nil {
+		return
+	}
+
+	var f interface{}
+	err := json.Unmarshal(*ext, &f)
+
+	if err != nil {
+		return
+	}
+
+	if features, ok := f.(map[string]interface{})["features"]; ok {
+		if ft, ok := features.(map[string]interface{}); ok {
+			for k, v := range ft {
+				if activated, ok := v.(map[string]interface{})["activated"]; ok {
+					a.clientFeatureStatusMap[k] = FeatureTimestamp{
```

Comment:
This is not thread safe and will inevitably lead to a panic or unexpected behavior. To fix this, you will need to protect read/write access with a lock of some kind.

I'm confused as to why the server needs to keep track of a feature from an incoming request? Why is each request not evaluated on it's own merit?

Further, Prebid Server is hosted in many large clusters. Such that encountering a single request with a feature enablement will only affect 1 out of many servers.